### ISSUE TITLE
The LocalizedControlType is 'none' for customized UserControl on Windows

### ISSUE THREAD

**Issue by jane-mao** [AvaloniaUI/Avalonia#17423]
### Describe the bug

The LocalizedControlType is 'none' for customized UserControl on Windows, please see the below screenshot.
<img width="1574" alt="Avalonia-Sample-LocalizedControlType-None" src="https://github.com/user-attachments/assets/aff24e72-d23d-44f1-9d3a-a08de2eab09f">

### To Reproduce

1. Build and run the attached the GetStartedApp.zip.
2. Use inspect.exe to check the above sample application, click on the OK button.

[GetStartedApp.zip](https://github.com/user-attachments/files/17613550/GetStartedApp.zip)

### Expected behavior

The LocalizedControlType is a valid value, but got `none`.

### Avalonia version

11.1.0

### OS

Windows

**Comment by maxkatz6** [comment]
UserControl should have AutomationControlType=Custom set on them.

**Comment by maxkatz6** [comment]
Avalonia automation is missing [UserControlAutomationPeer](https://github.com/AvaloniaUIOU/Xpf/blob/153ea6c04bf13f17b9bb072028fdb25c844a7d02/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/UserControlAutomationPeer.cs#L15) class as it seems. It should be added, and UserControl should return it from the OnCreateAutomationPeer method.

**Comment by jane-mao** [comment]
Hi @maxkatz6, any plan for the fix of this issue?

**Comment by timunie** [comment]
@jane-mao help wanted label means a PR is welcome. So if you want to give it a go, you're welcome to do so.
